business_requirement/models/business.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+# © 2015
+# Elico Corp, contributor: Eric Caudal, Alex Duan, Xie XiaoPeng
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api, fields, models
+
+
+class BusinessRequirement(models.Model):
+    _name = "business.requirement"
+    _description = "Business Requirement"
+
+    sequence = fields.Char(
+        'Sequence',
+        readonly=True,
+        copy=False,
+    )
+    name = fields.Char(
+        'Name',
+        required=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    description = fields.Char(
+        'Description', required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    business_requirement = fields.Text(
+        'Customer Story',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    scenario = fields.Text(
+        'Scenario',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    gap = fields.Text(
+        'Gap',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    drop = fields.Boolean(
+        string='Drop?',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help="""Determine whether this business requirement is to be kept
+        or dropped."""
+    )
+    categ_id = fields.Many2one(
+        comodel_name='business.requirement.category',
+        string='Category',
+        ondelete='restrict',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        selection="_get_states",
+        string='State',
+        default='draft',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    business_requirement_ids = fields.One2many(
+        comodel_name='business.requirement',
+        inverse_name='parent_id',
+        string='Sub Business Requirement',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    parent_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Parent',
+        ondelete='set null',
+        domain="[('id', '!=', id)]"
+    )
+    level = fields.Integer(
+        compute='_get_level',
+        string='Level',
+        store=True
+    )
+    change_request = fields.Boolean(
+        string='Change Request?',
+        default=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
```

Comment:
This is still a WIP. We need to add the project link in BR (Required) and add a smart button in project to see all BR.
Later for CRM purpose we will add the link of the project in the opportunity.